Makefile

```diff
@@ -3,52 +3,56 @@
 EMACS ?= emacs
 SRC = $(filter-out %-pkg.el, $(wildcard *.el reporters/*.el))
 ELC = $(SRC:.el=.elc)
+AUTOLOADS = google-translate-autoloads.el
 CASK ?= cask
 PKG_DIR := $(shell $(CASK) package-directory)
 FEATURES = $(wildcard features/*.feature)
 VERSION = 0.12.0
-TARGET_DIR = google-translate-$(VERSION)
 
-all: test marmalade tag
+%.elc: %.el autoloads
+	$(EMACS) -Q -batch -L . --eval \
+	"(let ((default-directory (expand-file-name \".cask\" default-directory))) \
+	   (require 'package) \
+	   (normal-top-level-add-subdirs-to-load-path))" \
+	-f package-initialize -f batch-byte-compile $<
+
+all: cask $(ELCS) autoloads
+
+autoloads: $(AUTOLOADS)
+
+$(AUTOLOADS): $(ELCS)
+	$(EMACS) -Q -batch -L . --eval \
+	"(progn \
+	   (require 'package) \
+	   (normal-top-level-add-subdirs-to-load-path) \
+	   (package-generate-autoloads \"phpactor\" default-directory))"
+
+cask: .cask/installed
+
+.cask/installed: Cask
+	$(CASK) install
+	@touch .cask/installed
 
 test:
 	$(MAKE) unit-test
 	$(MAKE) ecukes
 
-unit-test:
+unit-test: .cask $(ELCS)
 	$(CASK) exec ert-runner
 
 $(PKG_DIR):
-	Cask
-	$(CASK) install
 	touch $@
 
 ecukes:
 	$(CASK) exec ecukes --reporter magnars --script $(FEATURES) --no-win
 
-marmalade: marmalade-tar marmalade-upload marmalade-rm
-
-marmalade-tar:
-	mkdir $(TARGET_DIR)
-	cp google-translate-core-ui.el $(TARGET_DIR)
-	cp google-translate-core.el $(TARGET_DIR)
-	cp google-translate-default-ui.el $(TARGET_DIR)
-	cp google-translate-query-auto-complete.el $(TARGET_DIR)
-	cp google-translate-smooth-ui.el $(TARGET_DIR)
-	cp google-translate.el $(TARGET_DIR)
-	cp README.md $(TARGET_DIR)
-	cp google-translate-pkg.el $(TARGET_DIR)
-	tar -cf google-translate-$(VERSION).tar $(TARGET_DIR)
-
-marmalade-upload:
-	marmalade-upload -u atykhonov google-translate-$(VERSION).tar || true
-
-marmalade-rm:
-	rm google-translate-$(VERSION).tar
-	rm -rf $(TARGET_DIR)
-
```

Comment:
marmalade is not working now

https://github.com/nicferrier/elmarmalade/issues/158